src/Interpreters/Context.cpp

```diff
@@ -5016,10 +5018,14 @@ void Context::addQueryParameters(const NameToNameMap & parameters)
         query_parameters.insert_or_assign(name, value);
 }
 
-void Context::addBridgeCommand(std::unique_ptr<ShellCommand> cmd) const
+void Context::addBackgroundShellCommand(std::unique_ptr<ShellCommand> cmd) const
 {
-    std::lock_guard lock(shared->mutex);
-    shared->bridge_commands.emplace_back(std::move(cmd));
+    shared->background_active_shell_commands.addCommand(std::move(cmd));
+}
+
+void Context::terminateBackgroundShellCommand(pid_t pid) const
```

Comment:
```suggestion
void Context::removeShellCommand(pid_t pid) const
```

Reply:
How about naming it just `removeShellCommand` and write a comment here that shell command is waited in the its destructor and killed depending on `terminate_in_destructor_strategy`.

Reply:
Have simplified naming in the class and its contents. 
But keep more detailed naming in `context`, as for me it is the place where it won't be superfluous